include/sisl/grpc/rpc_server.hpp

```diff
@@ -118,8 +117,10 @@ class GrpcServer : private boost::noncopyable {
 
     // generic service methods
     bool run_generic_handler_cb(const std::string& rpc_name, boost::intrusive_ptr< GenericRpcData >& rpc_data);
+    void run_generic_completion_cb(const std::string& rpc_name, boost::intrusive_ptr< GenericRpcData >& rpc_data);
     bool register_async_generic_service();
-    bool register_generic_rpc(const std::string& name, const generic_rpc_handler_cb_t& rpc_handler);
+    bool register_generic_rpc(const std::string& name, const generic_rpc_handler_cb_t& rpc_handler,
+                              const generic_rpc_completed_cb_t& done_handler = nullptr);
```

Comment:
Do you think is it better to have 2 callbacks or just a single callback with handle vs completion as some sort of flag. It can avoid you doing std::unordered_map of std::pair, but need all caller back implentations to update their signatures.